{{cookiecutter.project_slug}}/_/ci-services/definitions/.gitlab-ci.yml

```diff
@@ -41,20 +41,40 @@ stages:
                  -f "deployment/application/Dockerfile" .
   - docker push "${IMAGE}"
 
+.generate-secrets:
+  before_script:
+  - POSTGRESQL_PASSWORD=$(< /dev/urandom tr -dc A-Za-z0-9 | head -c16)
+  - kubectl -n ${TARGET} get secret ${APPLICATION_NAME} ||
+    kubectl -n ${TARGET} create secret generic ${APPLICATION_NAME} --from-literal=DJANGO_SECRET_KEY=$(< /dev/urandom tr -dc A-Za-z0-9 | head -c50) \
+    --from-literal=DJANGO_DATABASE_URL=postgres://{{ cookiecutter.project_slug.replace('-', '_') }}:${POSTGRESQL_PASSWORD}@${DATABASE_NAME}/{{ cookiecutter.project_slug.replace('-', '_') }}
+  - kubectl -n ${TARGET} get secret ${DATABASE_NAME} ||
+    kubectl -n ${TARGET} create secret generic ${DATABASE_NAME} --from-literal=POSTGRESQL_PASSWORD=${POSTGRESQL_PASSWORD} --from-literal=POSTGRESQL_USER={{ cookiecutter.project_slug.replace('-', '_') }} --from-literal=POSTGRESQL_DATABASE={{ cookiecutter.project_slug.replace('-', '_') }}
+
 .deploy:
   stage: deploy
-  image: docker.io/appuio/oc:v3.9
+  extends: generate-secrets
+  environment:
+    url: "${KUBE_URL}/console/project/{{ cookiecutter.project_slug }}{% if cookiecutter.environment_strategy == 'dedicated' %}-${CI_ENVIRONMENT_NAME}{% endif %}/overview"
+  variables:
+    TARGET: "{{ cookiecutter.project_slug }}{% if cookiecutter.environment_strategy == 'dedicated' %}-${CI_ENVIRONMENT_NAME}{% endif %}"
+    APPLICATION_NAME: "application{% if cookiecutter.environment_strategy == 'shared' %}-${CI_ENVIRONMENT_NAME}{% endif %}"
+    DATABASE_NAME: "database{{ cookiecutter.project_slug }}{% if cookiecutter.environment_strategy == 'shared' %}-${CI_ENVIRONMENT_NAME}{% endif %}"
+  image: docker.io/appuio/oc:v3.11
   script:
-  - oc project ${TARGET}
-  - oc get secret postgres ||
-    oc process -f deployment/postgres-secrets.yaml |
-    oc apply --overwrite -f -
-  - oc get secret application ||
-    oc process -f deployment/application-secrets.yaml |
-    oc apply --overwrite -f -
   - oc tag "${SOURCE}/${CI_PROJECT_NAME}:${CI_COMMIT_SHA}"
            "${TARGET}/${CI_PROJECT_NAME}:${CI_COMMIT_SHA}"
```

Comment:
Will we still need a re-tagging with the `CI_COMMIT_SHA` with the "shared" strategy? Probably not, hm?